server/viame_tasks/tasks.py

```diff
@@ -15,8 +15,52 @@
 from typing import Dict, List
 
 
+def get_available_gpus() -> Dict[str, str]:
+    """
+    Return the list of available GPUs.
+
+    The return of this function is a dictionary that maps each GPU's UUID to its index.
+    """
+    # Get info from nvidia-smi, in the format: index, gpu-uuid
+    # Use this format instead of list due to required equal signs
+    command = "nvidia-smi --query-gpu=index,uuid --format=csv,noheader,nounits"
+
+    process = Popen(command, stderr=PIPE, stdout=PIPE, shell=True)
+    stdout_bytes, _ = process.communicate()
+
+    if process.returncode != 0:
+        return {}
+
+    # Create a list of the form [..., [index, uuid], ...]
+    stdout_lines = [line.strip() for line in stdout_bytes.decode().split("\n") if line]
+    stdout_tuples: List[List] = [line.split(", ") for line in stdout_lines]
+
+    uuid_to_index_dict = {uuid: index for index, uuid in stdout_tuples}
+    return uuid_to_index_dict
+
+
+def get_gpu_environment() -> Dict[str, str]:
+    """Get environment variables for using CUDA enabled GPUs."""
+    env = os.environ.copy()
+
+    # Default gpu index to the first one
+    job_gpu_index = "0"
+    available_gpus = get_available_gpus()
+
+    gpu_uuid = os.environ.get("WORKER_GPU_UUID")
+    if gpu_uuid:
+        job_gpu_index = available_gpus.get(gpu_uuid) or job_gpu_index
+
+    if available_gpus:
+        env["CUDA_VISIBLE_DEVICES"] = str(job_gpu_index)
+
+    return env
```

Comment:
If you use the above tool, you don't have to filter by uuid yourself, you can use excludeUUID.

Reply:
I don't think that's true, since `excludeUUID` will exclude the specified UUIDs from the list of returned ids (indices). What I'll probably end up doing is using `GPUtil.getGPUs`, and filtering that list by UUID.

Reply:
Got it.